src/gc/impl/conservative/gc.d

```diff
@@ -193,6 +208,8 @@ class ConservativeGC : GC
             cstdlib.free(gcx);
             gcx = null;
         }
+        version (Posix)
+            Gcx.instance = gcx;
```

Comment:
I think this pointer should be reset before gcx is destroyed. BTW: as only members of gcx are affected, how about moving this and the initialization into Gcx.Dtor and initialize, respectively?

Reply:
The lock is in `ConservativeGC`, and we need to refer to it from the callbacks. Accessing `ConservativeGC` from `Gcx` which seemed to me like a layering violation, but I see now there is precedent.